### PR TITLE
Add !votefor command and slight refactoring

### LINKED ISSUE #24: Discord bot !votefor command
We have to add a `!votefor` command to the Discord bot, for the client-side functionality of our voting event.

## Rules
1. Each fellow has a single vote to cast. Once cast, their vote is final, i.e. no takebacks.
2. Votes are weighted so a fellow's received votes are added to their own vote. For example, if a fellow has received five votes, their vote is worth six votes.
3. Each fellow must cast their vote. If a fellow has not voted by the end of the voting period, they cannot be picked as a winner.
4. Only fellows can vote and be voted for, that is, MLH staff, mentors, maintainers, sponsors are excluded. In fairness, we three should be excluded as well.
5. Vote numbers remain secret until the results are announced. We give access to an MLH admin so they can monitor that vote numbers are legit.

## Implementation

The Discord bot get a new command `!votefor @Discord username` and three possible answers.

```
// command

!votefor kendevops

// happy path

<bot> Your vote has been recorded. Thanks for voting!

// edge case 1

<bot> The records show you already voted!

// edge case 2

<bot> This person is not a fellow. Only fellows can receive votes! 
```

The DB gets two new columns `votes_available` (one or zero) and votes_received (zero and above) for each fellow.

When a fellow issues the command, the bot makes a GET request to the backend to check for votes available.

- Happy path: If voter has a vote to cast, the bot makes a POST requests to update (1) the voter's `votes_available` to zero, and (2) the votee's `votes_received` to whatever votes the votee has plus the voter's `votes_received` + 1. The bot confirms successful vote to the voter.

- Edge cases: If the voter has no vote to cast or if the votee is not in the fellows table or if the voter and the votee are the same, the bot reports why the vote cannot be cast.

When the contest ends, we look in the DB for three fellow with the highest number of votes, provided they have zero votes available.

### PR DESCRIPTION
**Pull request by ivov**
Closes #24.

Includes slight refactoring for easier future maintenance.

### CODE REVIEW

File: discord-bot/commands/votefor.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		console.log(recipientGithubId);
+
+		// if (["ivov", "kendevops", "shu8"].includes(recipientGithubId)) {
+		// 	message.channel.send(
+		// 		"Oh that fellow is a voting event organizer! Please vote for another fellow!"
+		// 	);
+		// 	return;
+		// }
+
+		if ([MLH_STAFF_GITHUB_IDS].includes(recipientGithubId)) {
```
Comment (shu8) [inline discord-bot/commands/votefor.js]: Should this be inside `[` brackets if it's already an array? (and the other few)

**Review by shu8 — COMMENTED** [review]
LGTM! just noticed this one thing, not sure if it might be a bug or not?

File: discord-bot/commands/votefor.js (hunk 1 above)
Comment (ivov) [reply discord-bot/commands/votefor.js]: Nice catch! Yeah, these bracketed checks would have failed tomorrow during the test run. Will fix!

**Review by shu8 — APPROVED** [review]
Awesome!

**Comment by shu8** [comment]
Merged through CLI, probably need to wait for GitHub to come back up properly before it's detected

https://www.githubstatus.com/

**Pull request closed without merging**